### DISCUSSION TITLE
Nested writes array connection

### DISCUSSION THREAD

**Discussion by mguldner** [prisma/prisma#10668] [Q&A] [answered]
Consider a school class with students, evaluations and grades for theses evaluations.
Here is my relation schema :

````
model Student {
  id            Int         @id @default(autoincrement())
  grades        Grade[]
}

model Evaluation {
  id            Int         @id @default(autoincrement())
  name          String
  grades        Grade[]
}

model Grade {
  id           Int        @id @default(autoincrement())
  student      Student    @relation(fields: [studentId], references: [id], onDelete: Cascade)
  studentId    Int        @map(name: "student_id")
  evaluation   Evaluation @relation(fields: [evaluationId], references: [id], onDelete: Cascade)
  evaluationId Int        @map(name: "evaluation_id")
  value        Int
}
````

On creation of an evaluation, I'd like to create as many grades for this evaluation as there are students, in one transaction (in order to perform rollback if any issue appears).
But I don't find any way to do it with [nested writes](https://www.prisma.io/docs/concepts/components/prisma-client/transactions).

Here is what I have for now :
````
const studentsId = [1, 2, 3, 4]

const evaluation = await prisma.evaluation.create({
        data: {
          name: "Evaluation",
          grades: {
            create: [
              value: 0,
              student: {
                connect: {id : studentId}  // I need to connect to each student of students array, not everytime the same !
              }
            ]
          }
        },
      });
````

I can't find a way to do this in the documentation or over the internet.
Bad search skills or missing feature ?

Thanks in advance :)

**Comment by mguldner** [comment, marked as answer]
Resolved: seems to be able to pass a previously-generated object to the transaction :
````
const gradesData = students.map((studentOfClass) => 
        ({
          value: 0,
          student: {
            connect: { id: studentOfClass.id}
          }
        })
      )

const evaluation = await prisma.evaluation.create({
        data: {
          name: "Evaluation",
          grades: {
            create: gradesData
          }
        },
      });
````

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
